include/bolgenos-ng/irq.hpp

```diff
@@ -30,7 +30,7 @@ typedef void (*irq_handler_t)(irq_t);
 * Max IRQ line that is configured and supported. The constant is
 *	inetnded to be used for **PIC configuration.
 */
-#define MAX_IRQ_NUMBER (0x2f)
+using max_irq_number = integral_constant<size_t, 0x2f>;
```

Comment:
since `max_irq_number` is in `irq::` NS, it's obviously a maximal number of IRQs. Rename it to `max_number` or `last_line` or something else.